packages/klighd-core/src/bookmarks/bookmark.tsx

```diff
@@ -0,0 +1,87 @@
+/** @jsx html */
+import { html } from "snabbdom-jsx"; // eslint-disable-line @typescript-eslint/no-unused-vars
+import { Command, CommandExecutionContext, CommandReturn, isViewport, Point, TYPES, ViewportAnimation } from "sprotty";
+import { Action } from "sprotty/lib/base/actions/action";
+import { inject } from "inversify";
+import { BookmarkPanel } from "./bookmark-panel";
+
+const ANIMATE_BOOKMARK = true;
+
+export class CreateBookmarkAction implements Action {
+    static readonly KIND = 'create-bookmark';
+    readonly kind = CreateBookmarkAction.KIND;
+}
+
+export class CreateBookmarkCommand extends Command {
+    static readonly KIND = CreateBookmarkAction.KIND;
+
+    constructor(@inject(TYPES.Action) protected action: CreateBookmarkAction) {
+        super();
+    }
+
+    execute(context: CommandExecutionContext): CommandReturn {
+        const model = context.root
+        if (isViewport(model)) {
+            BookmarkPanel.addBookmark("MakeBookmark" + BookmarkPanel.getLenghtBookmarks(), <
+                                        div>Bookmark {BookmarkPanel.getLenghtBookmarks()}</div>, 
```

Comment:
Is it necessary to store the UI (JSX) together with the data? This makes it harder to find the right location to change the rendering, which is naturally expected in the panel render method.

Reply:
I think this is also now solved by thee BookmarkRegistry